api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -57,19 +70,127 @@ describe('assessmentsRouter', () => {
+                question_id: 1,
+                title: 'A relational database management system',
+                description: '',
+                sort_order: 1,
+                correct_answer: true,
+              },
+            ],
+            correct_answer_id: 1,
+            max_score: 1,
+            sort_order: 1,
+          },
+        ],
+      };
+      const programAssessment = {
+        id: programAssessmentId,
+        program_id: 1,
+        assessment_id: curriculumAssessmentId,
+        available_after: '2023-02-06',
+        due_date: '2023-02-10',
+      };
+
+      const assessmentSubmission = {
+        id: 2,
+        submission_id: 2,
+        assessment_id: programAssessmentId,
+        principal_id: participantPrincipalId,
+        assessment_submission_state: 'Graded',
+        score: 10,
+        opened_at: '2023-02-09 12:00:00',
+        submitted_at: '2023-02-09 13:23:45',
+        responses: [
+          {
+            id: 1,
+            answer_id: 4,
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 1,
+            score: 1,
+          },
+        ],
+      };
+      const response: {
+        curriculum_assessment: CurriculumAssessment;
+        program_assessment: ProgramAssessment;
+        assessment_submission: AssessmentSubmission;
+      } = {
+        curriculum_assessment: curriculumAssessment,
+        program_assessment: programAssessment,
+        assessment_submission: assessmentSubmission,
       };
+
+      mockGetCurriculumAssesmentBasedOnRole.mockResolvedValue(response);
+      mockPrincipalId(facilitatorPrincipalId);
       appAgent
-        .get(`/${exampleAssessmentId}/submissions/${exampleSubmissionId}`)
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+
         .expect(200, itemEnvelope(response), err => {
+          expect(mockGetCurriculumAssesmentBasedOnRole).toHaveBeenCalledWith(
+            facilitatorPrincipalId,
+            programAssessmentId,
+            submissionId
+          );
+
           done(err);
         });
     });
+    it('should respond with a bad request error if given an invalid assessment id', done => {
+      const assessmentId = 'test',
+        submissionId = 1;
+
+      appAgent
+        .get(`/${assessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
+    });
+    it('should respond with a bad request error if given an invalid submission id ', done => {
+      const assessmentId = 1,
+        submissionId = 'test';
+
+      appAgent
+        .get(`/${assessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
+    });
+
+    it('should respond with an internal server error if an error', done => {
+      const assessmentId = 1,
+        submissionId = 1;
+      mockGetCurriculumAssesmentBasedOnRole.mockRejectedValue(new Error());
+      appAgent
+        .get(`/${assessmentId}/submissions/${submissionId}`)
+        .expect(500, done);
+    });
   });
```

Comment:
We need additional tests here. Something along the lines of:

- "should show a participant their submission information for an in-progress assessment without including the correct answers"
- "should show a participant their submission information for an ungraded submitted assessment without including the correct answers"
- "should show a participant their submission information for a graded submitted assessment including the correct answers"
- "should respond with a NotFoundError if the assessment ID was not found in the database"
- "should respond with a NotFoundError if the submission ID was not found in the database"
- "should respond with an UnauthorizedError if the logged-in principal ID is not the same as the principal ID of the submission ID and is not the principal ID of the program facilitator"